### PR TITLE
pgwire: convert a few more auth tests to datadriven input

### PR DESCRIPTION
**Pull request by knz**
TestPGWire, RootUserAuth and TestUserAuth had been converted already.
This completes by converting UnicodeUserAuth and also using that to
exercise username normalization.

The patch also converts TestPGWireNonexistentUser in insecure mode (a
test already existed for secure mode).

It also makes `connect` tests print out the name of the current db
after login, and uses that to replace TestPGWireDBName.

Release note: None

### CODE REVIEW

File: pkg/sql/pgwire/testdata/auth/empty_hba [hunk 1]
```diff
@@ -97,3 +97,29 @@ DROP USER testuser_nocert
...
+subtest unicode_user_and_username_normalization
+
+sql
+CREATE USER ὀδυσσεύς WITH PASSWORD '蟑♫螂';
+----
+ok
+
+# Verify auth fails without password.
+#
+# Also verify that the username gets normalized: the username
```
Comment (mjibson) [inline pkg/sql/pgwire/testdata/auth/empty_hba]: Do we also need to test here that the non-normalized username (i.e., the one provide initially) also works?

**Review by madelynnblue — APPROVED** [review]

File: pkg/sql/pgwire/testdata/auth/empty_hba (hunk 1 above)
Comment (knz) [reply pkg/sql/pgwire/testdata/auth/empty_hba]: Good idea. Done.

**Comment by knz** [comment]
TFYR!

bors r+

**Comment by knz** [comment]
(simplifies fixing #31113)

**Pull request merged**